src/track-type-dependency.js

```diff
@@ -9,8 +10,21 @@ export default function trackTypeDependency(typeName) {
   types[typeName] = true;
 }
 
+export function trackFieldDependency(typeName, fieldName) {
+  if (!tracking) {
+    return;
+  }
+
+  if (!fields[typeName]) {
+    fields[typeName] = {};
+  }
+
+  fields[typeName][fieldName] = true;
```

Comment:
What about just appending to an array? 

```js
fields[typeName] = [];
fields[typeName].concat(fieldName);
```

Any advantages to either one?

Might make building `trackedFields` simpler?